### DISCUSSION TITLE
TCP server leads to memory leaking

### DISCUSSION THREAD

**Discussion by ChivenZhang** [libuv/libuv#3711] [General]
The code below in main function has been found memory leak with windows 10 task monitor. When i post 10000 requests to server with JMeter, it happens. Expect fixing !!! Thanx

<hr/>
auto loop = (uv_loop_t*)malloc(sizeof(uv_loop_t));<br/>uv_loop_init(loop);<br/><br/>uv_tcp_t server;<br/>uv_tcp_init(loop, &server);<br/><br/>struct sockaddr_in addr;<br/>uv_ip4_addr("0.0.0.0", 8888, &addr);<br/><br/>int result = uv_tcp_bind(&server, (const struct sockaddr*)&addr, 0);<br/>if (result) {<br/>	printf("Could not bind server to port %d. Reason: %s\n", 8888, uv_strerror(result));<br/>	return;<br/>}<br/><br/>result = uv_listen((uv_stream_t*)&server, 128,<br/>	[](uv_stream_t* server, int status) {<br/>		if (status < 0) {<br/>			fprintf(stderr, "New connection error %s\n", uv_strerror(status));<br/>			return;<br/>		}<br/>		uv_tcp_t* client = (uv_tcp_t*)malloc(sizeof(uv_tcp_t));<br/>		uv_tcp_init(server->loop, client);<br/>		if (uv_accept(server, (uv_stream_t*)client) == 0) {<br/>			uv_close((uv_handle_t*)client, nullptr);<br/>		}<br/>		else {<br/>			uv_close((uv_handle_t*)client, nullptr);<br/>		}<br/>	});<br/>if (result) {<br/>	printf("Could not listen. Reason: %s\n", uv_strerror(result));<br/>	return;<br/>}<br/><br/>uv_run(loop, UV_RUN_DEFAULT);<br/>uv_loop_close(loop);

<hr/>

* **Version**: 1.43.0
* **Platform**: Windows 10

**Comment by vtjnash** [comment]
You never free memory in this snippet, so that is not libuv's fault (libuv does not often allocate or free memory for you)
